lib/network_utils.ml

```diff
@@ -246,21 +250,61 @@ module Sysfs = struct
 			Result.Ok devices.(0)
 		with _ -> Result.Error (Parent_device_of_vf_not_found, "Can not get parent device for " ^ pcibuspath)
 
-	let device_index_of_vf parent_device pcibuspath =
+	let get_child_vfs_sysfs_paths dev = 
 		try
-			let re = Re_perl.compile_pat "virtfn(\d+)" in
-			let device_path = getpath parent_device "device" in
-			let group = Sys.readdir device_path
+			let device_path = getpath dev "device" in
+			Result. Ok (
+				Sys.readdir device_path
 				|> Array.to_list
-				|> List.filter (Re.execp re) (* List elements are like "virtfn1" *)
-				|> List.find (fun x -> Xstringext.String.has_substr (Unix.readlink (device_path ^ "/" ^ x)) pcibuspath )
-				|> Re.exec_opt re
+				|> List.filter (Re.execp (Re_perl.compile_pat "virtfn(\d+)")) (* List elements are like "virtfn1" *)
+				|> List.map (Filename.concat device_path)
+			)
+		with _ -> Result.Error (Vf_sysfs_path_not_found, "Can not get child vfs sysfs paths for " ^ dev)
+
+	let device_index_of_vf parent_dev pcibuspath =
+		try
+			let open Rresult.R.Infix in
+			get_child_vfs_sysfs_paths parent_dev >>= fun paths ->
+			let group = 
+				List.find (fun x -> Xstringext.String.has_substr (Unix.readlink x) pcibuspath) paths
```

Comment:
`xcp-networkd` has been ported away from `Xstringext`.
Please do not use `Xstringext` and instead use 
```ocaml
Atring.String.is_infix ~affix:(Unix.readlink x)
```